packages/client/src/generation/TSClient/TSClient.ts

```diff
@@ -91,18 +76,23 @@ export class TSClient implements Generatable {
       generator.config.engineType = engineType
     }
 
+    const inlineSchema = (await readFile(schemaPath)).toString('base64')
+    const inlineSchemaHash = crypto.createHash('sha256').update(inlineSchema).digest('hex')
     const config: Omit<GetPrismaClientConfig, 'runtimeDataModel' | 'dirname'> = {
       generator,
       relativeEnvPaths,
-      sqliteDatasourceOverrides,
       relativePath: pathToPosix(path.relative(outputDir, path.dirname(schemaPath))),
       clientVersion: this.options.clientVersion,
       engineVersion: this.options.engineVersion,
       datasourceNames: datasources.map((d) => d.name),
       activeProvider: this.options.activeProvider,
-      dataProxy: this.options.dataProxy,
       postinstall: this.options.postinstall,
       ciName: ciInfo.name ?? undefined,
+      inlineDatasources: datasources.reduce((acc, ds) => {
+        return (acc[ds.name] = { url: ds.url }), acc
+      }, {} as GetPrismaClientConfig['inlineDatasources']),
+      inlineSchema,
+      inlineSchemaHash,
```

Comment:
Simplified datasource information needed by the dataproxy. In a follow up PR we will warn users if they have a hardcoded URL in their schema, as it would appear here otherwise.

Reply:
Maybe we could also have this warning surfaced in the Prisma VS Code extension / `prisma validate` 🤔